presto-main/src/test/java/io/prestosql/operator/scalar/FunctionAssertions.java

```diff
@@ -324,132 +323,55 @@ public void assertInvalidFunction(String projection)
         }
     }
 
-    public void assertInvalidFunction(String projection, StandardErrorCode errorCode, String messagePattern)
+    public void assertInvalidFunction(String projection, ErrorCodeSupplier errorCode, String message)
     {
-        try {
-            evaluateInvalid(projection);
-            fail("Expected to throw a PrestoException with message matching " + messagePattern);
-        }
-        catch (PrestoException e) {
-            try {
-                assertEquals(e.getErrorCode(), errorCode.toErrorCode());
-                assertTrue(e.getMessage().equals(messagePattern) || e.getMessage().matches(messagePattern), format("Error message [%s] doesn't match [%s]", e.getMessage(), messagePattern));
-            }
-            catch (Throwable failure) {
-                failure.addSuppressed(e);
-                throw failure;
-            }
-        }
+        assertPrestoExceptionThrownBy(() -> evaluateInvalid(projection))
+                .hasErrorCode(errorCode)
+                .hasMessage(message);
```

Comment:
Previously, the code allowed equality or pattern matching for error messages.
Now, with the fluent API allows us to treat them separately, but for callers of `assertInvalidFunction`, this is not a drop-in replacement.

Reply:
Yes, based on my (somewhat incomplete) visual verification, all of the callers used messages. I renamed all of the parameters to reflect that.

I think that the previous behavior of allowing both is not a good API design.